src/object/object_manager.cpp

```diff
@@ -162,7 +162,14 @@ CObject* CObjectManager::CreateObject(Math::Vector pos, float angle, ObjectType
     params.pos = pos;
     params.angle = angle;
     params.type = type;
-    params.power = power;
+    
+    float min = 0;
+    float max = 100;
+    if (params.type == OBJECT_POWER || params.type == OBJECT_ATOMIC) //prevent creation of overcharged or negatively charged power cells
+    {
+        max = 1;
+    }
+    params.power = Math::Clamp(power,min,max);
```

Comment:
This code is almost the same, I think it would be a good idea to extract into a function/method.